app/src/test/java/com/losandroides/learn/presentation/MainViewModelTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.losandroides.learn.presentation
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.losandroides.learn.domain.ItemsUseCase
+import com.losandroides.learn.domain.model.Item
+import com.losandroides.learn.framework.coVerifyOnce
+import com.losandroides.learn.framework.relaxedMockk
+import io.mockk.coEvery
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.*
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class MainViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private val testDispatcher = StandardTestDispatcher()
+    private val testCoroutineScope = TestScope(testDispatcher)
+
+    private val itemsUseCase: ItemsUseCase = relaxedMockk()
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @After
+    fun cleanup() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `GIVEN some items WHEN MainViewModel is initialized THEN ItemsUseCase is called`() {
+        val items = buildItems()
+        coEvery { itemsUseCase() } returns items
+
+        testCoroutineScope.runTest {
+            val viewModel = buildMainViewModel()
+
+            coVerifyOnce { itemsUseCase() }
+
+            viewModel.viewState.value shouldBeEqualTo MainViewModel.ViewState.Content(items)
+        }
+    }
+
+    private fun buildMainViewModel() = MainViewModel(
```

Comment:
I am trying to pass the `dispatcher` by the `MainViewModel` constructor. But I don't achieve to make it work. Could anyone help me with that? Maybe @RubenGonz2?

Reply:
The test is working fine in my local. I I have restarted the action again, just in case.

Reply:
It llks like the problem is related to the Kluent library:

```
com.losandroides.learn.presentation.MainViewModelTest > GIVEN some items WHEN MainViewModel is initialized THEN ItemsUseCase is called FAILED
    org.amshove.kluent.internal.ComparisonFailedException at MainViewModelTest.kt:44
```

Reply:
Thanks for reviewing it, @RubenGonz2.
Sometimes the test passes.
But I guess it is because the threads can have that case.
I don't feel Kluent is the issue.
I have made some changes to move forward with the solution but it stills failing.